### PR TITLE
fix: pdfx build fails

### LINKED ISSUE #318: build fails
ERROR: ../../../../.pub-cache/hosted/pub.dartlang.org/pdfx-2.1.1/lib/src/renderer/web/platform.dart:2:8: Error: Not found: 'dart:html'
ERROR: import 'dart:html' as html;
ERROR:        ^
ERROR: ../../../../development/flutter/packages/flutter_web_plugins/lib/src/navigation/js_url_strategy.dart:13:8: Error: Not found: 'dart:html'
ERROR: import 'dart:html' as html;
ERROR:        ^
ERROR: ../../../../development/flutter/packages/flutter_web_plugins/lib/src/navigation/url_strategy.dart:6:8: Error: Not found: 'dart:html'
ERROR: import 'dart:html' as html;
ERROR:        ^
ERROR: ../../../../development/flutter/packages/flutter_web_plugins/lib/src/navigation/utils.dart:5:8: Error: Not found: 'dart:html'
ERROR: import 'dart:html';
ERROR:        ^
ERROR: ../../../../.pub-cache/hosted/pub.dartlang.org/pdfx-2.1.1/lib/src/renderer/web/platform.dart:3:8: Error: Not found: 'dart:js'
ERROR: import 'dart:js' as js;
ERROR:        ^
ERROR: ../../../../.pub-cache/hosted/pub.dartlang.org/pdfx-2.1.1/lib/src/renderer/web/pdfjs.dart:22:8: Error: Not found: 'dart:html'
ERROR: import 'dart:html';
ERROR:        ^
ERROR: ../../../../.pub-cache/hosted/pub.dartlang.org/js-0.6.4/lib/js.dart:8:1: Error: Not found: 'dart:js'
ERROR: export 'dart:js' show allowInterop, allowInteropCaptureThis;
ERROR: ^
ERROR: ../../../../.pub-cache/hosted/pub.dartlang.org/js-0.6.4/lib/js_util.dart:8:1: Error: Not found: 'dart:js_util'
ERROR: export 'dart:js_util';
ERROR: ^
ERROR: ../../../../.pub-cache/hosted/pub.dartlang.org/pdfx-2.1.1/lib/src/renderer/web/pdfjs.dart:23:8: Error: Not found: 'dart:js_util'
ERROR: import 'dart:js_util';
ERROR:        ^
ERROR: ../../../../.pub-cache/hosted/pub.dartlang.org/pdfx-2.1.1/lib/src/renderer/web/document/page.dart:2:8: Error: Not found: 'dart:html'

### LINKED ISSUE #319: 2.1.1 error messages
2.1.1 has error messages and will not build:

* Messages.kt: (30, 1): Class 'Messages' is not abstract and does not implement abstract member public abstract fun unregisterTexture(message: Pigeon.UnregisterTextureMessage): Unit defined in dev.flutter.pigeon.Pigeon.PdfxApi
* Messages.kt: (286, 5): 'unregisterTexture' overrides nothing
* Error: Not found: 'dart:html'
* Error: Not found: 'dart:js'

and many more.

### PR DESCRIPTION
**Pull request by ekasetiawans**
fix #318 
fix #319

### CODE REVIEW

File: packages/auto_animated/example/pubspec.lock [hunk 1]
```diff
@@ -164,5 +164,5 @@ packages:
     source: hosted
     version: "0.3.3"
 sdks:
-  dart: ">=2.17.0-0 <3.0.0"
+  dart: ">=2.17.1 <3.0.0"
```
Comment (mx1up) [inline packages/auto_animated/example/pubspec.lock:167]: unrelated change?

Comment (ekasetiawans) [reply packages/auto_animated/example/pubspec.lock:167]: sorry, it happened because I did `flutter pub update`.

**Comment by GuillaumeFerron** [comment]
Hi guys, any ETA one this one? :)

**Comment by DattatreyaReddy** [comment]
Any updates on this?

**Comment by ekasetiawans** [comment]
I don't know why this is not merged yet

**Comment by SuryaSharpsell** [comment]
@mx1up - Any idea when this fix will be merged ?

**Comment by mx1up** [comment]
@SuryaSharpsell no, i am not involved with this project..

**Comment by hs029** [comment]
I tried this PR but still get build failure.

----
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':pdfx:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
-----

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.4, on macOS 12.4 21F79 darwin-arm, locale en-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] IntelliJ IDEA Community Edition (version 2022.1.2)
[✓] VS Code (version 1.69.1)
[✓] Proxy Configuration
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!

**Comment by SergeShkurko** [comment]
@ekasetiawans thanks for you contribution!

**Comment by mzdm** [comment]
@SergeShkurko I think you forgot to release a new version on pub.dev 🙏
